client/lib/domains/get-domain-type-text.js

```diff
@@ -26,7 +26,7 @@ export function getDomainTypeText(
 	switch ( domain.type ) {
 		case domainTypes.MAPPED:
 			if ( context === domainInfoContext.PAGE_TITLE ) {
-				return __( 'Connected domain' );
+				return __( 'Connected Domain' );
```

Comment:
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.